lte/gateway/c/core/oai/test/amf/test_amf.cpp

```diff
@@ -462,6 +462,30 @@ TEST(test_amf_nas5g_pkt_process, test_amf_registration_reject_msg) {
       decode_reg_rej.m5gmm_cause.m5gmm_cause);
 }
 
+TEST(test_amf_nas5g_pkt_process, test_amf_service_reject_message) {
```

Comment:
can you do this similar to the test case `test_amf_registration_reject_msg`, i.e., encode, then decode and compare the success of these ops as well as compare final decoded data is the same as the encoded one?

Reply:
Done